front/lib/api/assistant/jit_actions.ts

```diff
@@ -47,6 +57,129 @@ export async function isJITActionsEnabled(
+  for (const m of conversation.content.flat(1)) {
+    if (
+      isContentFragmentType(m) &&
+      isSupportedPlainTextContentType(m.contentType) &&
+      m.contentFragmentVersion === "latest"
+    ) {
+      if (m.fileId) {
+        files.push({
+          fileId: m.fileId,
+          title: m.title,
+          contentType: m.contentType,
+        });
+      }
+    } else if (isAgentMessageType(m)) {
+      for (const a of m.actions) {
+        if (isTablesQueryActionType(a)) {
+          if (a.resultsFileId && a.resultsFileSnippet) {
+            files.push({
+              fileId: a.resultsFileId,
+              contentType: "text/csv",
+              title: getTablesQueryResultsFileTitle({ output: a.output }),
+            });
+          }
+        }
+      }
+    }
+  }
+
+  // Fetch the snippets
+  const fileIds = files.map((f) => f.fileId);
+  const fileResources = await FileResource.fetchByIds(auth, fileIds);
+  for (const f of files) {
+    const fileResource = fileResources.find((fr) => fr.sId === f.fileId);
+    if (fileResource && fileResource.snippet) {
+      f.snippet = fileResource.snippet;
+    }
+  }
+
+  return files;
+}
+
+export async function getJITActions(
+  auth: Authenticator,
+  { conversation }: { conversation: ConversationType }
+): Promise<AgentActionConfigurationType[]> {
+  const actions: AgentActionConfigurationType[] = [];
+
+  if (await isJITActionsEnabled(auth)) {
+    const files = await listFiles(auth, { conversation });
+    if (files.length > 0) {
+      const filesUsableForJIT = files.filter((f) => !!f.snippet);
+
+      if (filesUsableForJIT.length > 0) {
+        // Get the datasource view for the conversation.
+        const dataSource = await DataSourceResource.fetchByConversationId(
+          auth,
+          conversation.id
+        );
+
+        if (!dataSource) {
+          logger.warn(
+            {
+              conversationId: conversation.sId,
+              fileIds: filesUsableForJIT.map((f) => f.fileId),
+              workspaceId: conversation.owner.sId,
+            },
+            "No datasource found for conversation when trying to get JIT actions."
+          );
+          return [];
+        }
+
+        const dataSourceView = (
+          await DataSourceViewResource.listForDataSources(auth, [dataSource])
+        ).find((dsv) => dsv.isDefault());
+
+        if (!dataSourceView) {
+          logger.warn(
+            {
+              conversationId: conversation.sId,
+              fileIds: filesUsableForJIT.map((f) => f.fileId),
+              workspaceId: conversation.owner.sId,
+            },
+            "No default datasource view found for conversation when trying to get JIT actions"
+          );
+
+          return [];
+        }
+
+        // Check tables for the table query action.
+        const filesUsableAsTableQuery = filesUsableForJIT.filter(
+          (f) => f.contentType == "text/csv" // TODO: there should not be a hardcoded value here
```

Comment:
We have a type guard for all table like mimeType

Reply:
What I mean is that I think we want to have a "table compatible mime type list" (which we don't have afaik)

Reply:
Indeed it seems we don't. Let's add it in files.ts ?

Reply:
Not yet, I'd like to do a sweep that includes also fixing the way we decide text vs tables in upload.ts and upsert.ts then (as well as some stuff in connectors).
Adding a task to "uniformize how we decide to handle as table vs text.